diesel/src/pg/connection/result.rs

```diff
@@ -48,6 +48,12 @@ impl PgResult {
                 ))
             }
             _ => {
+                // "clearing" the connection by polling result till we get a null.
+                // this will indicate that the previous command is complete and
+                // the same connection is ready to process next command.
+                // https://www.postgresql.org/docs/current/libpq-async.html
+                while conn.get_next_result()?.is_some() {}
```

Comment:
We want to return the orginal error here, so we want to avoid using `?` at all. Something like

```suggestion
                while conn.get_next_result().map(Option::is_some).unwrap_or(true) {}
```

(although this will recursively call the same function (`PgResult::new`) again and potentially run this loop there again. I believe that should be fine as this should stop at some layer wen `get_next_result()` returns `None`)

Reply:
@weiznich Is the default value `true` intended? Wouldn't it keep the loop running even after some error? Am I missing something?

Reply:
Yes `true` is the correct default value as far as I can tell. The libpq documentation states that `PQgetResult` (which is called internally by this function) must be called till it returns `null`. It does not talk about any errors there, so I assume that means we literally need to call it again till it returns null.

That by itself seems reasonable as [`get_next_result()`](https://github.com/diesel-rs/diesel/blob/master/diesel/src/pg/connection/raw.rs#L119) calls `PQgetResult` (which cannot return an error) and then `PgResult::new()` (which can return an error). So the only way `get_next_result()` would ever return an error is by hitting basically the current code block. We would like to ignore that as the first error message is relevant.